### PR TITLE
[APB-3386][MP,AO] Split submit service into separate routes for personal and business

### PR DESCRIPTION
**Pull request by mariannepearson0**

### CODE REVIEW

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala [hunk 1]
```diff
@@ -94,9 +94,14 @@ class AgentLedDeAuthController @Inject()(
       handleGetSelectServicePage(agentConfirmationForm("cancel-authorisation.error.business-service.required")))
   }
 
-  def submitSelectService: Action[AnyContent] = Action.async { implicit request =>
+  def submitSelectPersonalService: Action[AnyContent] = Action.async { implicit request =>
     ifShowDeAuthFlag(
-      handleSubmitSelectService(agentConfirmationForm("cancel-authorisation.error.business-service.required")))
+      handleSubmitSelectServicePersonal(agentConfirmationForm("cancel-authorisation.error.business-service.required")))
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala]: the error message here uses `business`

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsInvitationController.scala [hunk 2]
```diff
@@ -92,8 +92,12 @@ class AgentsInvitationController @Inject()(
     handleGetSelectServicePage()
   }
 
-  val submitSelectService: Action[AnyContent] = Action.async { implicit request =>
-    handleSubmitSelectService()
+  val submitSelectPersonalService: Action[AnyContent] = Action.async { implicit request =>
+    handleSubmitSelectServicePersonal(agentConfirmationForm("error.business-service.required"))
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentsInvitationController.scala]: same as above

File: it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationsITSAControllerJourneyISpec.scala [hunk 3]
```diff
@@ -17,8 +17,8 @@ class AgentInvitationsITSAControllerJourneyISpec extends BaseISpec with AuthBeha
   lazy val controller: AgentsInvitationController = app.injector.instanceOf[AgentsInvitationController]
 
   "POST /agents/select-service" should {
```
Comment (sureshhmrc) [inline it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationsITSAControllerJourneyISpec.scala]: not a biggie , but could you fix this message as well as its generic now ? :)

File: it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationsVATControllerJourneyISpec.scala [hunk 4]
```diff
@@ -22,8 +22,8 @@ class AgentInvitationsVATControllerJourneyISpec extends BaseISpec with AuthBehav
   implicit val hc: HeaderCarrier = HeaderCarrier(sessionId = Some(SessionId(Random.alphanumeric.take(10).toString())))
 
   "POST /agents/select-service" should {
```
Comment (sureshhmrc) [inline it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationsVATControllerJourneyISpec.scala]: same as above

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/BaseInvitationController.scala [hunk 5]
```diff
@@ -736,15 +724,19 @@ abstract class BaseInvitationController(
 
   def selectServiceCall: Call = routes.AgentsInvitationController.showSelectService()
 
-  def submitServiceCall: Call = routes.AgentsInvitationController.submitSelectService()
+  def submitServicePersonalCall: Call = routes.AgentsInvitationController.submitSelectPersonalService()
+
+  def submitServiceBusinessCall: Call = routes.AgentsInvitationController.submitSelectPersonalService()
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentinvitationsfrontend/controllers/BaseInvitationController.scala]: is this right ?

File: app/uk/gov/hmrc/agentinvitationsfrontend/views/agents/SelectServicePageConfig.scala [hunk 6]
```diff
@@ -38,4 +39,6 @@ case class SelectServicePageConfig(basketFlag: Boolean, featureFlags: FeatureFla
     map.toSeq
   }
 
+  val submitCall: Call = routes.AgentsInvitationController.submitSelectPersonalService()
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentinvitationsfrontend/views/agents/SelectServicePageConfig.scala]: would this work incase of de-auth journey ?  or are we overriding this value for de-auth journey any where ?

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala (hunk 1 above)
Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala]: Yeah i think that right, that form is the business form for when there are errors in the page and it redirects back to get SelectService which takes business form. Perhaps should use default or named parameters

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/BaseInvitationController.scala (hunk 5 above)
Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/BaseInvitationController.scala]: no, fixed 👍

File: it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationsITSAControllerJourneyISpec.scala (hunk 3 above)
Comment (mariannepearson0) [reply it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationsITSAControllerJourneyISpec.scala]: 👍

File: app/uk/gov/hmrc/agentinvitationsfrontend/views/agents/SelectServicePageConfig.scala (hunk 6 above)
Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentinvitationsfrontend/views/agents/SelectServicePageConfig.scala]: no, fixed 👍

File: it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationsVATControllerJourneyISpec.scala (hunk 4 above)
Comment (mariannepearson0) [reply it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationsVATControllerJourneyISpec.scala]: 👍

File: app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala (hunk 1 above)
Comment (mariannepearson0) [reply app/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentLedDeAuthController.scala]: I've made it a default parameter to avoid confusion

**Pull request merged**